### PR TITLE
Source Salesforce: update regex pattern, specify format to date-time

### PR DESCRIPTION
**Pull request by arsenlosenko**
## What
User is not able to set `start_date` properly during setup of Source Salesforce. See [slack thread](https://airbytehq-team.slack.com/archives/C02U9R3AF37/p1678818865149109) for more context

## How
Remove regex pattern that was used previously, specify `date-time` as format, use it for validation.

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Comment by erohmensing** [comment]
Hm, is this a breaking change for connectors that were set up with the `"2021-07-25" format? 

Maybe adding the `format: date` was a breaking change to users who were configured using the date-time format too?

**Review by bazarnov — APPROVED** [review]

**Comment by erohmensing** [comment]
@arsenlosenko do you know the answer to my question above?

**Comment by arsenlosenko** [comment]
@erohmensing sorry for the late response, I don't have an answer to your question right now, let me check if this breaks anything once again. Last night I tested sync locally with format both set to `date` and `date-time` and it worked, but I'll test this once again for good measure

**Comment by erohmensing** [comment]
@arsenlosenko thanks! By trying it with both do you mean you kept the `format: whatever` the same in the spec, and tried passing both values? Or you changed the spec each time?

**Comment by arsenlosenko** [comment]
@erohmensing I changed the spec each time, but let me try to pass different values with the same format value in spec, this way we could see if this affects anything

**Comment by erohmensing** [comment]
@artem1205 unfortunately testing with the spec changed won't reproduce the actual behavior since only one version of the spec can be canonical at once. 

We should also test `check` in addition to the sync as I think this is where it was failing

**Comment by artem1205** [comment]
cc @arsenlosenko

**Comment by arsenlosenko** [comment]
@erohmensing @artem1205 I am still
testing check and sync with different formats of date in spec, I'll get back to you once I have the results

**Comment by erohmensing** [comment]
Ope, sorry for the incorrect ping @artem1205!

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/spec.yaml [hunk 1]
```diff
@@ -47,11 +47,11 @@ connectionSpecification:
         Enter the date in the YYYY-MM-DD format. Airbyte will replicate the data added on and after this date. If this field is blank, Airbyte will replicate all data.
       type: string
       pattern: >-
-        ^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z|[0-9]{4}-[0-9]{2}-[0-9]{2}$
+        ^([0-9]{4}-[0-9]{2}-[0-9]{2}(T[0-9]{2}:[0-9]{2}:[0-9]{2}Z)?)$
       examples:
         - "2021-07-25"
```
Comment (grubberr) [inline airbyte-integrations/connectors/source-salesforce/source_salesforce/spec.yaml:52]: Do we need to remove this example ?

**Review by grubberr — CHANGES_REQUESTED** [review]

**Comment by grubberr** [comment]
@arsenlosenko do we need fix this https://github.com/airbytehq/airbyte/blob/master/docs/integrations/sources/salesforce.md?plain=1#L51
?

**Comment by arsenlosenko** [comment]
@grubberr I want to check if we can use different date formats, but as you indicated, in docs there is only one specified date format, compared to examples in spec, which are in different format. I am not that familiar with this connector, and don't know if this date/date-time will affect the performance of the source. 
If we are relying only on description of this field in docs, then yes, we can set format to `date`, remove example of datetime from spec, and update regex pattern.

**Comment by bnchrch** [comment]
Sorry for a delayed review.

I wanted to see what current values we had for start_date in prod:

image

Unfortunately we have both date, datetime and null.

Double checking the regex provided shows that it matches both cases
image
image

So it looks like we should keep the permissive regex and extra example

**Review by bnchrch — APPROVED** [review]
Approving as it seems like this is the correct way to do pattern validation at this fime. (See previous comment)

 Before publishing and merging @arsenlosenko you can confirm that spec, check and sync are working for both example formats?

**Comment by arsenlosenko** [comment]
@bnchrch thanks for the review and clarification 👍
And yes,I will check if connector performs as expected with both date formats before publishing and and merging it

**Comment by arsenlosenko** [comment]
@bnchrch I have tested setup of connector locally with different date formats, using only `date-time` format in spec, works as expected.
Screenshot 2023-03-16 at 10 50 07
Screenshot 2023-03-16 at 10 43 33

Check, spec and sync works as well, will proceed to publish and merge after tests and CI are green

**Review by grubberr — APPROVED** [review]

**Review by grubberr — APPROVED** [review]

**Pull request merged**